### PR TITLE
Add test for new rounded money format functions

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Unit tests in support of https://github.com/civicrm/civicrm-core/pull/12626

Before
----------------------------------------
Untested

After
----------------------------------------
tested

Technical Details
----------------------------------------
Note that there are also some protected functions added. I have left those to be teste via this wrapper as they are not yet exposed

Comments
----------------------------------------
@mattwire @seamuslee001 ping

### CODE REVIEW

File: tests/phpunit/CRM/Utils/MoneyTest.php [hunk 1]
```diff
@@ -44,4 +46,29 @@ public function subtractCurrenciesDataProvider() {
...
+   */
+  public function testFormatLocaleNumericRoundedByCurrency() {
+    $result = CRM_Utils_Money::formatLocaleNumericRoundedByCurrency(8950.3678, 'NZD');
+    $this->assertEquals('8,950.37', $result);
+  }
+
+  /**
+   * Test rounded by currency function.
+   *
+   * This should be formatted according to European standards - . thousand separator
```
Comment (seamuslee001) [inline tests/phpunit/CRM/Utils/MoneyTest.php]: @eileenmcnaughton this comment doesn't seem to match the text which seems to be using `,` as a thousand separator and `.` as decimal

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Utils/MoneyTest.php]: doesn't it     $this->assertEquals('8.950,37', $result);

Comment (seamuslee001) [reply tests/phpunit/CRM/Utils/MoneyTest.php]: it looks to me to be `$this->assertEquals('8,950.37', $result);`

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Utils/MoneyTest.php]: Ah yeah - what I had locally wasn't properly pushed

Comment (seamuslee001) [reply tests/phpunit/CRM/Utils/MoneyTest.php]: ok test looks right now

**Comment by seamuslee001** [comment]
This only affects tests so the only judge that really matters is Jenkins, adding merge on pass

**Pull request merged**